velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -3837,3 +3837,69 @@ TEST_F(DateTimeFunctionsTest, fromUnixtimeDouble) {
   });
   assertEqualVectors(expected, actual);
 }
+
+TEST_F(DateTimeFunctionsTest, toISO8601TestTimestampWithTimezone) {
+  static const int kSecondsInMinute = 60;
+  static const int kSecondsInHour = 3'600;
+  static const int64_t kSecondsInDay = 86'400;
+  static const uint64_t kNanosInSecond = 1'000'000'000;
+
+  const auto toISO8601 = [&](std::optional<int64_t> timestamp,
+                             const std::optional<std::string>& timeZoneName) {
+    return evaluateWithTimestampWithTimezone<std::string>(
+        "to_iso8601(c0)", timestamp, timeZoneName);
+  };
+
+  // (A) wall clock passed to function = 1180149582s (UTC relative)
+  // (B) wall clock to output          = 1180149582s - 04:00 (local)
+  EXPECT_EQ(
+      "2007-05-25T23:19:42.000-04:00",
+      toISO8601(((long long)1180149582) * 1000, "America/New_York"));
```

Comment:
Why this long long cast? Is this to make sure there is no overflow because you are multiplying it with 1000 later. So best thing is to use the static cast to the type you want, in this case  int64_t.

Reply:
What about the subsecond precision value for the timestamp?
Can there be tests where you have something in the millisecondvalue range and not just full seconds?

Reply:
Good point. Ended up rewriting the cast by defining 1000 as a constant int64_t, to implicitly cast via arithmetic.